web-console/future-openshift/contextual-help/contextual-help.md

```diff
@@ -0,0 +1,85 @@
+  * If there’s something a user should do (that we cannot do for them), use the Hint pattern to describe or provide a shortcut (link, button) for taking that next step.
+  * When the next step has been completed, users should not see the hint again for that resource.
+* Use the Infopop pattern to provide information for things users may have to do, but which may not be obvious.
+  * Use this pattern when we’re not sure when (or even if) a user may need the information.
+  * This pattern is useful when users may need to do something outside of the product UI to accomplish a goal.
+* Target help to intermediate users
+  * The majority of our users will be intermediate users.
+  * Being a beginner is not a permanent condition -- beginning users become intermediate users.
+  * Intermediate and advanced users will understand basic concepts and functions.
+* Onboarding and general information
+  * **This document does not specifically address onboarding or general information.**
+  * We should be looking at other ways to provide onboarding, or more general information to beginning users.
+  * This information should be available to novice users, but should be out of the way of intermediate and advanced users.
+
+## Hint Pattern
+The hint pattern is similar to an informational inline notification, but with additional emphasis (blue highlight). The intention is to use this for specific use cases where we can anticipate the next step a user should take, or to introduce new features. The hint pattern involves a one-step reminder, explanation, or call to action.
+
+### Potential Use cases
+1. To provide contextual help on using **new features**
+2. To provide hints on a flow/ the **next steps** a user should be thinking about.
+
+#### New Features
+![hint pattern - new features](img/hint-new-features.png)
+
+#### Next Steps
+![hint pattern - next steps](img/hint-next-step.png)
+
+### Design
+#### General
+![hint pattern](img/hint-general.png)
+* Hint area is shown in context of where it applies
+  * This may be above the page title for cases where the hints apply to the entire page of the application.
+  * This may be below a specific tab or area on a page for cases where the hints only apply to a portion.
+* Hint area is highlighted in blue
+* Hint must have a title (h3 style)
+* Hint must have some body text (body-copy style)
+* There can optionally be external links within the body text
+* There can optionally be a primary action below the body text
+* Title, body text, and primary action are left aligned
+* For cases without a primary action, there can optionally be an X in the top right corner to dismiss the message
+
+#### Colors
+* Font: `pf-blue-600`
+* Background: `pf-blue-50`
+* Primary Action: Standard PatternFly button colors
+
+#### Spacing
+![hint pattern spacing](img/hint-spacing.png)
+* `18px` border between content and Background
+* `30px` space between the body text and primary button (when used)
+
+## Infopop Pattern
```

Comment:
For KNI, we are proposing using the popover on certain statuses in the table. Not sure if it makes sense to include as an option, but figured I would at least mention it!
https://github.com/openshift/openshift-origin-design/pull/179

FYI @beanh66 @andybraren

Reply:
I feel like what 's here and what @andybraren has are compatible. I could crossreference Andy's design, if that'd be helpful.

Andy, I wonder in your design if the title on the popover is necessary? I suggested the headless popover by default because I wasn't sure that it would always be necessary/helpful, but also it could be difficult to come up with a pithy title in some cases. I'm could be convinced either way.

Reply:
I agree, we seem compatible. I'm not sure the Status Popover design is quite "contextual help" or an "infopop" though, so cross-referencing might not be necessary.

Headless infopops by default makes sense. For Status Popovers I lean toward including titles partially because it allows the paragraph text to flow the full width of the popover (headless puts extra white space along the right-hand side under the X) but also because KubeVirt has at least one scenario where the status cell text (`Running*`) would be different enough from the popover's title (`Guest agent missing`, seen in #179) that including the title would make it easier to quickly understand the `*`.

That's admittedly a pretty niche example (and could change), but I suspect we'll encounter others as KNI & KubeVirt continue to add additional status types and messages.